great_expectations/datasource/fluent/fluent_base_model.py

```diff
@@ -240,6 +245,12 @@ def dict(
             exclude_none=exclude_none,
             skip_defaults=skip_defaults,
         )
+        if config_provider:
+            logger.info(
+                f"{self.__class__.__name__}.dict() - substituting config values"
+            )
+            _recursively_set_config_value(result, config_provider)
```

Comment:
Does this resolve all the config values? Is this what we want to do in a call to `dict`? I'm curious about the use case and if we call this to do functional things in code?

Reply:
We do this when we need to pass kwargs to things like the `BatchSpec` or an `ExecutionEngine`

Reply:
I see, thanks for the examples.